Sources/PortalSwift/Core/Mpc/PortalMpc.swift

```diff
@@ -820,26 +804,83 @@ public class PortalMpc {
     return mpcShare
   }
 
-  private func parseRecoveryInput(data: Data) async throws -> PortalMpcGenerateResponse {
+  private func formatShares(sharesJSONString: String) async throws -> PortalMpcGenerateResponse {
+    let formattedSharesString = self.mobile.MobileFormatShares(sharesJSONString)
+
+    guard let jsonData = formattedSharesString.data(using: .utf8) else {
+      throw MpcError.unableToDecodeShare
+    }
+
     do {
-      return try self.decoder.decode(PortalMpcGenerateResponse.self, from: data)
-    } catch {
-      // If the backup isn't in the new format, try to get it in the old format and convert to the new format
-      let mpcShare = try decoder.decode(MpcShare.self, from: data)
-      guard let shareString = String(data: data, encoding: .utf8) else {
-        throw MpcError.unableToDecodeShare
+      let result = try JSONDecoder().decode(FormatShareResponse.self, from: jsonData)
+
+      if let error = result.error {
+        throw MpcError.unexpectedErrorOnRecover("Error formatting shares: \(error.message)")
       }
-      var backupShares: PortalMpcGenerateResponse = [:]
 
-      backupShares["SECP256K1"] = PortalMpcGeneratedShare(
-        id: mpcShare.backupSharePairId ?? "",
-        share: shareString
-      )
+      guard let data = result.data else {
+        throw MpcError.unexpectedErrorOnRecover("No data returned from MobileFormatShares")
+      }
 
-      return backupShares
+      return data
```

Comment:
is this data in the format of PortalMpcGenerateResponse?

Reply:
Yupp! Otherwise there would have been compilation errors